src/stores/spaces/SpaceStore.ts

```diff
@@ -380,19 +385,31 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
         return false;
     }
 
-    public getSpaceFilteredRoomIds = (space: SpaceKey): Set<string> => {
+    // get all rooms in a space
+    // including descendant spaces
+    public getSpaceFilteredRoomIds = (space: SpaceKey, includeDescendantSpaces = true): Set<string> => {
         if (space === MetaSpace.Home && this.allRoomsInHome) {
             return new Set(this.matrixClient.getVisibleRooms().map(r => r.roomId));
         }
-        return this.spaceFilteredRooms.get(space) || new Set();
+
+        if (!includeDescendantSpaces) {
+            return this.spaceFilteredRooms.get(space) || new Set();
+        }
+        return flattenSpaceHierarchy(this.spaceFilteredRooms, this.childSpacesBySpace)(space);
     };
 
-    public getSpaceFilteredUserIds = (space: SpaceKey): Set<string> => {
+    public getSpaceFilteredUserIds = (space: SpaceKey, includeDescendantSpaces = true): Set<string> => {
         if (space === MetaSpace.Home && this.allRoomsInHome) {
             return undefined;
         }
-        if (isMetaSpace(space)) return undefined;
-        return this.spaceFilteredUsers.get(space) || new Set();
+        if (isMetaSpace(space)) {
+            return undefined;
+        }
+        if (!includeDescendantSpaces) {
+            return this.spaceFilteredUsers.get(space) || new Set();
+        }
+
+        return flattenSpaceHierarchy(this.spaceFilteredUsers, this.childSpacesBySpace)(space);
```

Comment:
The param name for `flattenSpaceHierarchy` is `spaceRoomMap` with type `SpaceRoomMap` yet we're giving it a Map from SpaceKey to Set of userIds, not roomIds.